logic/standalone/waterfall.py

```diff
@@ -0,0 +1,129 @@
+'''
+Logic module that can create waterfall objects.
+
+
+USAGE EXAMPLE:
+
+'''
+
+import logic.common.log_utils as log
+import logic.common.tiled_utils as tiled_utils
+
+#--------------------------------------------------#
+'''Variables'''
+
+TILELAYER_PREFIX = "_falls"
+
+DEFAULT_SORT_FG = "fg_tiles,15"
+DEFAULT_SORT_BG = "bg_tiles,5"
+DEFAULT_SORT = DEFAULT_SORT_BG
+
+DEFAULT_COLOR = "ffffff"
+
+OBJECTGROUP_PREFIX = "objects_waterfall_"
+
+
+
+#--------------------------------------------------#
+'''Public Functions'''
+
+def GetMetaDataFromBlockout(playdo):
+	'''Obtain the tile2d and various meta data for the waterfall, based on tilelayer name
+	  e.g. _falls => sort_value is DEFAULT_SORT, color is DEFAULT_COLOR
+	       _falls_fg15 => sort_value is "fg_tiles,15", color is DEFAULT_COLOR
+	       _falls_bg_bbccff => sort_value is DEFAULT_SORT_BG, color is "#bbccff/1f"
+	'''
+
+	''' =====[Pseudo-code]=====
+	list_meta_data = []
+	Get all tilelayer name
+	for tilelayer in list_all_tilelayer:
+		layer_name = tiled_utils.GetName ( tilelayer ) 
+		if layer_name doesn't start with TILELAYER_PREFIX : continue
+
+		layer_tuple = layer_name.split("_")
+		sort_value = _ProcessSort( layer_tuple[1] )
+		color_value = _ProcessColor( layer_tuple[2] )
+
+		list_meta_data.append( tilelayer.GetTile2D(), sort_value, color_value )
+
+	return list_meta_data
+	'''
+
+	return []
+
+
+def ScanForWaterFalls(list_meta_data):
+	'''...'''
+
```

Comment:
definitely fill in descriptions (the really descriptive kind, describe the input args and return value). Because that's what's going to convey the intent for your design